### PR TITLE
Define `MuchDecimal` mixin

### PR DESCRIPTION
**Pull request by jcredding**
This makes the `MuchDecimal` a mixin that can be used to define
attributes that are decimals stored as integers. This allows easily
setting up attributes that are decimals but store their values as
integers. The decimals can be used for presentation/display while
the integer can be used for math operations. This is particularly
useful when working with currency.

This brings in much-plugin and makes the main module `MuchDecimal`
a plugin. The plugin simply adds a class method
`decimal_as_integer` that allows setting up decimal as integer
attributes. The class method takes an attribute name and a set of
options. The source attribute and a precision are the possible
options. The source attributes specifies the accessor that the
integer will be written to. The precision is the number of decimal
places or how precise you want the integer to be. Both of these are
defaulted if they aren't provided.

The class method defines a reader and writer. The readers reads
from the source accessor expecting an integer and converting it to
its decimal using the precision. The writer expects a decimal and
converts it to an integer using the precision and writes it to the
source accessor. Both of these handle `nil` and non numeric values
appropriately.

Finally, this pulls the conversion logic, decimal to integer and
integer to decimal, into singleton methods on the main module.
This allows their logic to be used directly. This is particularly
useful for testing that an attribute is using much-decimal.

@kellyredding - Ready for review. I coped the edge case tests as-is from where we originally implemented this.

### CODE REVIEW

File: lib/much-decimal.rb [hunk 1]
```diff
@@ -1,5 +1,48 @@
...
+
+  module ClassMethods
+
+    def decimal_as_integer(attribute, options = nil)
+      options ||= {}
+      source    = options[:source] || "#{attribute}_as_integer"
+      precision = (options[:precision] || DEFAULT_PRECISION).to_i
+
+      define_method(attribute) do
+        integer_as_decimal = self.send(source)
```
Comment (kellyredding) [inline lib/much-decimal.rb]: @jcredding can we just call this local var `integer`?

File: lib/much-decimal.rb [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+        MuchDecimal.integer_to_decimal(integer_as_decimal, precision)
+      end
+
+      define_method("#{attribute}=") do |decimal_as_integer|
+        new_value = MuchDecimal.decimal_to_integer(decimal_as_integer, precision)
```
Comment (kellyredding) [inline lib/much-decimal.rb]: @jcredding likewise, can we just call the block arg `decimal` and the local var `integer`?  I'm just trying to keep it obvious the state of the data at each step.

**Comment by kellyredding** [comment]
@jcredding a couple minor var name comments - otherwise this looks good man 👎

File: test/unit/much_decimal_tests.rb [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      assert_equal 5.0,   subject.seconds
+    end
+
+    should "handle decimals that are less than 1" do
+      subject.seconds = 0.0001
+      assert_equal 1,      subject.ten_thousandth_seconds
+      assert_equal 0.0001, subject.seconds
+    end
+
+    should "handle decimals with to many significant digits by rounding" do
```
Comment (kellyredding) [inline test/unit/much_decimal_tests.rb]: @jcredding "... with _too_ many ..."
